Detectors/Upgrades/ALICE3/FCT/simulation/include/FCTSimulation/Detector.h

```diff
@@ -0,0 +1,181 @@
+
+#include <vector>                             // for vector
+#include "DetectorsBase/GeometryManager.h"    // for getSensID
+#include "DetectorsBase/Detector.h"           // for Detector
+#include "DetectorsCommonDataFormats/DetID.h" // for Detector
+#include "ITSMFTSimulation/Hit.h"             // for Hit
+#include "Rtypes.h"                           // for Int_t, Double_t, Float_t, Bool_t, etc
+#include "TArrayD.h"                          // for TArrayD
+#include "TGeoManager.h"                      // for gGeoManager, TGeoManager (ptr only)
+#include "TLorentzVector.h"                   // for TLorentzVector
+#include "TVector3.h"                         // for TVector3
+#include "FCTBase/FCTBaseParam.h"
+
+class FairVolume;
+class TGeoVolume;
+
+class TParticle;
+
+class TString;
+
+namespace o2
+{
+namespace fct
+{
+class GeometryTGeo;
+}
+} // namespace o2
+namespace o2
+{
+namespace fct
+{
+class FCTLayer;
+}
+} // namespace o2
+
+namespace o2
+{
+namespace fct
+{
+class FCTLayer;
+
+class Detector : public o2::base::DetImpl<Detector>
+{
+ public:
+  /// Name : Detector Name
+  /// Active: kTRUE for active detectors (ProcessHits() will be called)
+  ///         kFALSE for inactive detectors
+  Detector(Bool_t active);
+
+  /// Default constructor
+  Detector();
+
+  /// Default destructor
+  ~Detector() override;
+
+  /// Initialization of the detector is done here
+  void InitializeO2Detector() override;
+
+  /// This method is called for each step during simulation (see FairMCApplication::Stepping())
+  Bool_t ProcessHits(FairVolume* v = nullptr) override;
+
+  /// Registers the produced collections in FAIRRootManager
+  void Register() override;
+
+  /// Gets the produced collections
+  std::vector<o2::itsmft::Hit>* getHits(Int_t iColl) const
+  {
+    if (iColl == 0) {
+      return mHits;
+    }
+    return nullptr;
+  }
+
+  /// Has to be called after each event to reset the containers
+  void Reset() override;
+
+  /// Base class to create the detector geometry
+  void ConstructGeometry() override;
+
+  /// This method is an example of how to add your own point of type Hit to the clones array
+  o2::itsmft::Hit* addHit(int trackID, int detID, const TVector3& startPos, const TVector3& endPos,
+                          const TVector3& startMom, double startE, double endTime, double eLoss,
+                          unsigned char startStatus, unsigned char endStatus);
+
+  Int_t chipVolUID(Int_t id) const { return o2::base::GeometryManager::getSensID(o2::detectors::DetID::FCT, id); }
+
+  void EndOfEvent() override;
+
+  void FinishPrimary() override { ; }
+  virtual void finishRun() { ; }
+  void BeginPrimary() override { ; }
+  void PostTrack() override { ; }
+  void PreTrack() override { ; }
+
+  /// Returns the number of layers
+  Int_t getNumberOfLayers() const { return mNumberOfLayers; }
+
+  void buildBasicFCT(const FCTBaseParam& param);
+  void buildFCTV1();
+  void buildFCTFromFile(std::string);
```

Comment:
that could be `const&` as far as I can see from the implementation of this method

Reply:
I tried, but that returned an error so I left it as it is

Reply:
well, you need to change that here and in the source file of course. from the source file (where the actual implementation of the function is) I don't see why this cannot be done.
Note: The function signatures have to match, so you cannot have `std::string` in one place and `std::string const&` in the other